File: linkpreviewedittext/src/main/java/in/codeshuffle/linkpreviewedittext/listener/LinkPreviewListener.java
package in.codeshuffle.linkpreviewedittext.listener;

import in.codeshuffle.linkpreviewedittext.LinkInfo;

public interface LinkPreviewListener {
    void onLinkFound(String url);

    void onLinkPreviewFound(LinkInfo linkInfo, boolean fromCache);

    void onNoLinkPreview();

    void onLinkPreviewError(String errorMsg);
}

File: linkpreviewedittext/src/main/java/in/codeshuffle/linkpreviewedittext/listener/LinkPreviewListener.kt
package `in`.codeshuffle.linkpreviewedittext.listener

import `in`.codeshuffle.linkpreviewedittext.LinkInfo

interface LinkPreviewListener {
    fun onLinkFound(url: String)
    fun onLinkPreviewFound(linkInfo: LinkInfo?, fromCache: Boolean)
    fun onNoLinkPreview()
    fun onLinkPreviewError(errorMsg: String)
}